src/IMangrove.sol

```diff
@@ -34,26 +58,40 @@ interface IMangrove is HasMgvEvents {
     bytes32 s
   ) external;
 
+  ///@notice See {IERC20Permit-nonces}.
+  ///@param owner The address of the owner.
+  ///@return nonce The current nonce of the owner.
   function nonces(address owner) external view returns (uint nonce);
 
   // # Taker functions
 
-  function marketOrderByVolume(OLKey memory olKey, uint takerWants, uint takerGives, bool fillWants)
-    external
-    returns (uint takerGot, uint takerGave, uint bounty, uint fee);
-
+  ///@notice Performs a market order on a specified offer list taking offers up to a limit price.
+  ///@param olKey The offer list key given by (maker) `outbound_tkn`, (maker) `inbound_tkn`, and `tickSpacing`.
+  ///@param maxTick Must be `>= MIN_TICK` and `<= MAX_TICK`. The limit price the taker is ready to pay (the log base 1.0001 of the price).
+  ///@param fillVolume Must be `<= MAX_SAFE_VOLUME`. If `fillWants` is true, the amount of `olKey.outbound_tkn` the taker wants to buy; otherwise, the amount of `olKey.inbound_tkn` the taker wants to sell.
+  ///@param fillWants if true, the matching engine tries to get the taker all they want; otherwise, the matching engine tries to sell all that the taker gives (subject to price).
```

Comment:
```suggestion
  ///@param fillWants if true, the matching engine tries to get the taker all they want; otherwise, the matching engine tries to sell all that the taker gives. In both cases subject to the price limit.
```